.github/workflows/haskell.yml

```diff
@@ -0,0 +1,172 @@
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc: ["8.6.5"]
+        os: [ubuntu-latest, windows-latest]
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Install pkgconfiglite
+      if: matrix.os == 'windows-latest'
+      run: choco install pkgconfiglite
+
+    - name: Peek
+      if: matrix.os == 'windows-latest'
+      run: pkg-config --variable pc_path pkg-config
+
+    - name: Install libsodium
+      if: matrix.os == 'windows-latest'
+      run: |
+        curl -Ls https://download.libsodium.org/libsodium/releases/libsodium-1.0.18-mingw.tar.gz -o libsodium-1.0.18-mingw.tar.gz
+        tar zxvf libsodium-1.0.18-mingw.tar.gz
+        pkg_config_dir="$(pkg-config --variable pc_path pkg-config | cut -d ';' -f 1)"
+        export PKG_CONFIG_PATH="$(readlink -f libsodium-win64/lib/pkgconfig)"
+        echo "PKG_CONFIG_PATH=$PKG_CONFIG_PATH"
+        echo "::set-env name=PKG_CONFIG_PATH::$PKG_CONFIG_PATH"
+        ls -al "$PKG_CONFIG_PATH"
+        # cp "libsodium-win64/lib/pkgconfig/libsodium.pc" "$pkg_config_dir"
+        echo "==="
+        sed -i "s|/d/a/1/s/|$(pwd)/|g" libsodium-win64/lib/pkgconfig/libsodium.pc
+        cat libsodium-win64/lib/pkgconfig/libsodium.pc
+        echo "==="
+        echo "==="
+        pkg-config --libs libsodium
+        echo "==="
+        pwd
+        echo "==="
+        ls -al
+        echo "=== lib ==="
+        ls /d/a/cardano-node/cardano-node/libsodium-win64/lib
+        echo "=== include ==="
+        ls /d/a/cardano-node/cardano-node/libsodium-win64/include
+
+    - name: Configure to use libsodium
+      run: |
+        cat >> cabal.project.local <<EOF
+        package cardano-crypto-praos
+        flags: -external-libsodium-vrf
+        EOF
+
+    - name: Debugging
+      env:
+        BINARY_CACHE_REGION: ${{ secrets.BINARY_CACHE_REGION }}
+        BINARY_CACHE_THREADS: ${{ secrets.BINARY_CACHE_THREADS }}
+        BINARY_CACHE_URI: ${{ secrets.BINARY_CACHE_URI }}
+      run: |
+        echo "BINARY_CACHE_REGION: $BINARY_CACHE_REGION"
+        echo "BINARY_CACHE_THREADS: $BINARY_CACHE_THREADS"
+        echo "BINARY_CACHE_URI: $BINARY_CACHE_URI"
+
+    - uses: actions/setup-haskell@v1
+      id: setup-haskell
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: '3.2.0.0'
+
+    - name: Install build environment
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        sudo apt-get update
+        sudo apt-get -y install libsodium23 libsodium-dev
+        sudo apt-get -y install libsystemd0 libsystemd-dev
+        sudo apt-get -y remove --purge software-properties-common
+        sudo apt-get -y autoremove
+
+        cat >> cabal.project <<EOF
+        package cardano-crypto-praos
+          flags: -external-libsodium-vrf
+        EOF
+
+    - name: Cabal update
+      run: cabal update
+
+    - name: Select build directory
+      run: echo "::set-env name=CABAL_BUILDDIR::dist"
```

Comment:
Will this pick the current directory?  On windows I had problems with length of paths names, so I always build somewhere very close to the root, e.g. `C://d`.

Reply:
So far this has been working.  Shortening the `dist-newstyle` to `dist` was just enough to squeeze in under the path name limit.